### PR TITLE
Skip ipv4, ipv6 loopback addresses in peer dialing

### LINKED ISSUE #160: Panic using `--libp2p 0.0.0.0:port` option
### Description

Running `polygon-sdk server --libp2p 0.0.0.0:10001` fails without passing a `--nat`

### Your environment

* OS and version Ubuntu 20
* version of the Polygon SDK 96be90487c144f7b1589ac65aad3f741c81644e3
* branch that causes this issue `develop`

### Steps to reproduce

* Tell us how to reproduce this issue <br /> Run `polygon-sdk server --libp2p 0.0.0.0:10001` with a genesis file 
* Where the issue is, if you know <br /> `AddrInfoToString` seems to not like when there are two addresses in the libp2p address info. It's giving me two addresses, 

```go
// AddrInfoToString converts an AddrInfo into a string representation that can be dialed from another node
func AddrInfoToString(addr *peer.AddrInfo) string {
	fmt.Print(addr)
	if len(addr.Addrs) != 1 {
		panic("Not supported")
	}
	return addr.Addrs[0].String() + "/p2p/" + addr.ID.String()
}

```
Gives
```
{16Uiu2HAm7dBcv1avSV8TVFZQV4kUNYi5N1jHu7C4xH1gV58McZx2: [/ip4/192.168.1.248/tcp/10002 /ip4/127.0.0.1/tcp/10002]}

```

### Logs

```
panic: Not supported

goroutine 1 [running]:
github.com/0xPolygon/polygon-sdk/network.AddrInfoToString(0xc0005a04e0, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/network/server.go:595 +0x1f7
github.com/0xPolygon/polygon-sdk/network.NewServer(0x1582748, 0xc0002f4cb0, 0xc000100eb0, 0x0, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/network/server.go:138 +0xe71
github.com/0xPolygon/polygon-sdk/server.NewServer(0x1582748, 0xc0002f4b60, 0xc0002f8f80, 0x0, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/server/server.go:92 +0x5ef
github.com/0xPolygon/polygon-sdk/command/server.(*ServerCommand).Run(0xc00000ed08, 0xc000032260, 0x4, 0x4, 0x0)
	/home/julian/sx-network/polygon-sdk/command/server/server_command.go:187 +0x3de
github.com/mitchellh/cli.(*CLI).Run(0xc00013f7c0, 0x0, 0x0, 0x0)
	/home/julian/sx-network/polygon-sdk/vendor/github.com/mitchellh/cli/cli.go:255 +0xb7d
main.Run(0xc000032250, 0x5, 0x5, 0x0)
	/home/julian/sx-network/polygon-sdk/main.go:25 +0x151
main.main()
	/home/julian/sx-network/polygon-sdk/main.go:12 +0x7c
```

### Proposed solution

If you pass a `--nat`, this error disappears. I feel like this should not be a panic.

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

This PR fixes issue #160.

It adds support for skipping loopback ipv4 and ipv6 loopback addresses when dialing.
All permutations of the loopback addresses are covered.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [x] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

File: network/server.go [hunk 1]
```diff
@@ -590,10 +591,31 @@ func StringToAddrInfo(addr string) (*peer.AddrInfo, error) {
...
 func AddrInfoToString(addr *peer.AddrInfo) string {
-	if len(addr.Addrs) != 1 {
-		panic("Not supported")
+	// Regex used for matching loopback addresses (IPv4 and IPv6)
+	// This regex will match:
+	// /ip4/localhost/tcp/<port>
+	// /ip4/127.0.0.1/tcp/<port>
+	// /ip4/<any other loopback>/tcp/<port>
+	// /ip6/<any loopback>/tcp/<port>
+	loopbackRegex := regexp.MustCompile(
```
Comment (Kourin1996) [inline network/server.go]: You can use same parser every time. You might want to declare as global variable to avoid useless cost.

Comment (zivkovicmilos) [reply network/server.go]: Good call 👍 

Moved it out to a constant:
0bfea14

File: network/server.go [hunk 2]
```diff
@@ -588,12 +589,35 @@ func StringToAddrInfo(addr string) (*peer.AddrInfo, error) {
...
 func AddrInfoToString(addr *peer.AddrInfo) string {
-	if len(addr.Addrs) != 1 {
-		panic("Not supported")
+	dialAddress := ""
+
+	// Find an address that's not a loopback address
+	for _, address := range addr.Addrs {
+		if !loopbackRegex.MatchString(address.String()) {
+			// Not a loopback address, dial address found
+			dialAddress = address.String()
```
Comment (Kourin1996) [inline network/server.go]: Just comments
(1) Is it ok if you don't exit loop after immediately you find suitable address?

Comment (zivkovicmilos) [reply network/server.go]: 🤦‍♂️

Fixed
07ed343

**Review by lazartravica — APPROVED** [review]

**Review by Kourin1996 — APPROVED** [review]
LGTM 🚀

**Comment by lazartravica** [comment]
@zivkovicmilos please feel free to merge 👍

**Pull request merged**